web/client/plugins/QueryPanel.jsx

```diff
@@ -190,8 +191,9 @@ const tocSelector = createSelector(
         appliedFilterSelector,
         storedFilterSelector,
         (state) => state && state.query && state.query.isLayerFilter,
-        selectedLayerLoadingErrorSelector
-    ], (enabled, groups, settings, querypanelEnabled, layoutHeight, dashboardAvailable, appliedFilter, storedFilter, advancedToolbar, loadingError) => ({
+        selectedLayerLoadingErrorSelector,
+        (state) => state.layers && state.layers.selected && state.layers.selected[0]
```

Comment:
The query panel is used also in other contexts (e.g. in widgets ). So the selected layer is not always the layer in the query panel. 
Use the one in query.typeName (there should be a selector for it).

Reply:
Thanks. Just as a side note - this looks like a bit of limitation, `typeName` is always a layer name, but it's not specific to the layer instance. So it won't be possible to have different panels for different instances of the same layer (if such precise configuration will be ever required)